compiler/modules/types/core.bal

```diff
@@ -815,9 +815,59 @@ function comparableNillableList(Context cx, SemType t1, SemType t2) returns bool
     }
     ComparableMemo memo = { semType1: t1, semType2: t2 };
     cx.comparableMemo.add(memo);
-    // SUBSET need to iterate members when tuples are supported
-    // following relies on the fact `listMemberType(cx, NIL, ()) = NEVER`
-    boolean result = comparable(cx, listMemberType(cx, t1, INT), listMemberType(cx, t2, INT));
+    boolean result;
+    var members1 = listAllMemberTypes(cx, t1);
+    var members2 = listAllMemberTypes(cx, t2);
+    // listAllMemberTypes return () if list type is not atomic
+    if members1 != () && members2 != () {
+        result = false;
+        int currentIndex = 0;
+        var [currentMemberRange, currentMemberTy] = members2[currentIndex];
+        foreach var [range, ty] in members1 {
+            if currentMemberRange.max >= range.max {
+                result = updateComparableMemo(cx, memo, currentMemberTy, ty);
+                if result == false {
+                    return false;
+                }
+                continue;
+            }
+            while currentMemberRange.max <= range.max {
+                if currentMemberRange.max > range.min {
+                    result = updateComparableMemo(cx, memo, currentMemberTy, ty);
+                    if result == false {
+                        return false;
+                    }
+                }
+                if currentMemberRange.max == range.max {
+                    break;
+                }
+                currentIndex += 1;
+                if currentIndex >= members2.length() {
+                    memo.comparable = false;
+                    return false;
+                }
+                [currentMemberRange, currentMemberTy] = members2[currentIndex];
+            }
+        }
+        SemType lastMemberTy = members1[members1.length() - 1][1];
+        while currentIndex < members2.length() {
+            [_, currentMemberTy] = members2[currentIndex];
+            result = updateComparableMemo(cx, memo, lastMemberTy, currentMemberTy);
+            if result == false {
+                return false;
+            }
+            currentIndex += 1;
+        }
+    }
+    else {
+        result = comparable(cx, listMemberType(cx, t1, INT), listMemberType(cx, t2, INT));
+    }
+    memo.comparable = result;
+    return result;
+}
+
+function updateComparableMemo(Context cx, ComparableMemo memo, SemType t1, SemType t2) returns boolean {
```

Comment:
I don't understand what you are trying to achieve with this function. `comparable` memoizes.

Reply:
function removed (no longer needed)